src/store/matchedAudiences.ts

```diff
@@ -35,18 +35,9 @@ class MatchedAudienceStore {
     storage.set(StorageKeys.MATCHED_AUDIENCE_IDS, this.matchedAudienceIds);
   }
 
-  setMatchedAudiences(newlyMatchedAudiences: MatchedAudience[]) {
```

Comment:
Question on this change: if we move more logic out of the modules there is going to be lots of code in the main file, which will need to be broken up later - you could run the matching and pass in the matched audiences and the store would handle that.

The store can expose other util methods too - get all ids etc...

This then would be easier to test?

Reply:
Do you mean moving all the unmatching/version bumping behaviour to the store object?

I personally like to keep my (stateful) objects as dumb as I can while maintaining logic in self-contained (pure) functions.

Maybe you can elaborate a little bit on how it would facilitate the testing, so I get your idea a little better?

Reply:
I get why you would say that (theoretically), but in this case we have created a module which handles action on the audience state and encapsulated it into a class - I feel this is the correct solution for our use, as we need to ensure actions from various other parts of the codebase abide by sum rules, such as loading and saving to localstorage should not be done by consumers etc...

For interactions with this state, having methods is simple to test, as you can import the class, insert some state, trigger the methods and know what you need to expect.

The index file should not have all the logic, it should be a short description of the flow; getAudience, getPageviews, matchAudiences, ...

WDYT?

Reply:
I agree with keeping the index level high.
I also get your point. I'll try to move and encapsulate things in the store class.